src/components/InvalidVinMessage/InvalidVinMessage.js

```diff
@@ -0,0 +1,19 @@
+import InvalidVinMessage from './StyledInvalidVinMessage';
+import Typography from '../../../styles/Typography';
+import useStore from '../../lib/hooks/useStore';
+
+export default function DisplayInvalidVinMessage() {
+	const displayVinError = useStore(state => state.displayVinError);
+
+	if (displayVinError) {
+		return (
+			<InvalidVinMessage>
+				<Typography variant="h4" color="white" textAlign="center">
+					VIN Eingabe inkorrekt! Bitte überprüfe deine VIN und probiere es erneut.
+				</Typography>
+			</InvalidVinMessage>
+		);
+	} else {
+		return null;
+	}
+}
```

Comment:
Instead of returning null if displayVinError is false, you should ask for this outside the component and only render it when displayVinError is true.

Reply:
The components purpose is to render a message. The conditional rendering should be solved outside of this component. This component should only be rendered, if there is a reason to do so.

Instead of doing this inside of the component:

```
export default function DisplayInvalidVinMessage() {
	const vinValidity = useStore(state => state.vinValidity);
	if (!vinValidity) {
		return (
			<p>Message here</p>
		)
	}
}
```

The wrapping component should include this component conditionally.

```
return (
		<>
			{!vinValidity && <InvalidVinMessage />}
			<Modal />
			<CardGrid />
		</>
)
```

Reply:
Hi Alex, thanks for the explanation. I understood what Dominik and you now meant.
A question that arises from that rendering approach is though, that if an invalid Vin is entered for the first time and the component InvalidVinMessage is rendered the input field resets. On a second input of an invalid Vin it does not as it should. How do I keep the input-values of the input field on the first initial render of the error message if rendered outside the component on the index.js?